libraries/AP_HAL_Linux/RPIOUARTDriver.cpp

```diff
@@ -75,31 +73,8 @@ void RPIOUARTDriver::begin(uint32_t b, uint16_t rxS, uint16_t txS)
     _initialised = false;
     while (_in_timer) hal.scheduler->delay(1);
 
-   /*
-     allocate the read buffer
-   */
-   if (rxS != 0 && rxS != _readbuf_size) {
-       _readbuf_size = rxS;
-       if (_readbuf != NULL) {
-           free(_readbuf);
-       }
-       _readbuf = (uint8_t *)malloc(_readbuf_size);
-       _readbuf_head = 0;
-       _readbuf_tail = 0;
-   }
-
-   /*
-     allocate the write buffer
-   */
-   if (txS != 0 && txS != _writebuf_size) {
-       _writebuf_size = txS;
-       if (_writebuf != NULL) {
-           free(_writebuf);
-       }
-       _writebuf = (uint8_t *)malloc(_writebuf_size);
-       _writebuf_head = 0;
-       _writebuf_tail = 0;
-   }
+    _readbuf.set_size(rxS);
+    _writebuf.set_size(txS);
```

Comment:
Ok, I missed that it was enforcing a minimum of 1024.
I didn't understand what you are saying about other places.